bin/genome-env-next

```diff
@@ -0,0 +1,432 @@
+#!/bin/bash
+
+# if this was found via PATH as opposed to a path to executable
+THIS="$(basename "$0")"
+if test "$THIS" == "$0"
+then
+    # delegate to local version if it is found
+    if git rev-parse --is-inside-work-tree 1> /dev/null 2> /dev/null
+    then
+        WORK_TREE="$(git rev-parse --show-toplevel)"
+        NEW_THIS="${WORK_TREE}/bin/${THIS}"
+        if test -x "$NEW_THIS"
+        then
+            echo "delegating to $NEW_THIS" 1>&2
+            exec "$NEW_THIS" "$@"
+        fi
+    fi
+fi
+
+function usage {
+    sed 's/^\ \{4\}//' <<EOS
+    usage: $THIS [-hsuw] [command args]
+        See \`$THIS -h' for detailed help.
+EOS
+}
+
+function help {
+    sed 's/^\ \{4\}//' <<EOS
+    NAME - $THIS
+
+    DESCRIPTION
+    ===========
+
+    \`$THIS' is used as a wrapper command to setup an environment for \`genome'.  For most use cases no options need to be specified.  For example,
+
+        $THIS genome --help
+
+    If you know you will need to run several commands in succession you can start a new shell by doing,
+
+        $THIS \$SHELL
+
+    OPTIONS
+    =======
+
+    -h    Display this detailed help.
+
+    -M    Disable migration of database.
+
+    -S    Disable configuration of database.
+
+    -s db_snapshot_name
+          The name of the database snapshot to use instead of the default.
+
+    -U    Disable configuration of UR.
+
+    -u ur_repo
+          The path to a UR repo to use instead of the default.
+
+    -v    Enable verbose status messages.
+
+    -W    Disable configuration of Workflow.
+
+    -w workflow_repo
+          The path to a Workflow repo to use instead of the default.
+EOS
+}
+
+function main {
+    local TESTDBSERVER_DB_NAME TESTDBSERVER_DB_HOST TESTDBSERVER_DB_PORT \
+        TESTDBSERVER_DB_USER="genome" \
+        TESTDBSERVER_DB_PASS="mypassword"
+
+    local DISABLE_DB DISABLE_MIGRATIONS DISABLE_UR DISABLE_WF
+    local SNAPSHOT_NAME TOP UR_REPO VERBOSE WF_REPO
+
+    if git rev-parse --is-inside-work-tree 1> /dev/null 2> /dev/null
+    then
+        TOP="$(git rev-parse --show-toplevel)"
+    else
+        TOP="$PWD"
+    fi
+
+    export TESTDBSERVER_URL="https://apipe-test-db.gsc.wustl.edu"
+
+    parse_opts "$@"
+    shift $((OPTIND-1))
+
+    rebuild_meta_db
+    setup_test_db "$TOP"
+    setup_env "$TOP"
+
+    stderr ''
+    stderr "Genome Perl: $(genome-perl -e 'printf(qq(%vd\n), $^V)')"
+    stderr "Genome Prove: $(genome-prove --version)"
+    stderr "UR: $(wtf UR | head -n 1)"
+    stderr "Workflow: $(wtf Workflow | head -n 1)"
+    stderr "Genome: $(wtf Genome | head -n 1)"
+    stderr "Test DB Snapshot: $SNAPSHOT_NAME"
+    stderr "Test DB Name: $TESTDBSERVER_DB_NAME"
+    stderr ''
+
+    if [ $# -gt 0 ]; then
+        echo "$@"... 1>&2
+        exec "$@"
+    else
+        if [ -n "$BASH" ]; then
+            echo 'Press <Ctrl>-D or type `logout` to return to normal environment.'
+            echo ''
+            if test -z "$GE_ENV_NAME"
```

Comment:
I used to do `if [ ... ]; then` but noticed the [git][] bash scripts, e.g. [git-rebase.sh][], do,

```
if test ...
then
    ...
fi
```

I don't know any reason to do one or the other except that I find the bash man page frustrating to search.  So I switch to `if test`.  I've also become more of a fan of keeping "statements" on separate lines so I don't do the `; then` anymore.

[git]: https://github.com/git/git
[git-rebase.sh]: https://github.com/git/git/blob/master/git-rebase.sh